iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -68,7 +70,15 @@ public async Task RunSampleAsync(TimeSpan sampleRunningTime)
             try
             {
                 await InitializeAndOpenClientAsync();
-                await Task.WhenAll(SendMessagesAsync(cts.Token), ReceiveMessagesAsync(cts.Token));
+
+                if (_transportType.Equals(TransportType.Http1))
+                {
+                    await TaskHelper.WhenAllFailFast(SendMessagesAsync(cts.Token), ReceiveMessagesAsync(cts.Token), CompleteReceivedMessagesAsync(cts.Token));

```

Comment:
Tas.WhenAll() will wait until all supplied tasks complete before returning the exception, but I wanted the application to fail as soon as a non-recoverable exception was encountered; hence this fail fast implementation

Reply:
his is esp because the supplied tasks are long-running which will terminate only when cancelled explicitly or if a non-recoverable exception is encountered

Reply:
Why not `Task.WhenAny()`?